pybossa/cache/projects.py

```diff
@@ -121,10 +125,13 @@ def format_task(row, lock_users=[]):
     if filter_user_prefs:
         # construct task list for worker view
         params["assign_user"] = args["sql_params"]["assign_user"]
-        results = session.execute(text(sql), params)
         task_rank_info = []
-
         user_profile = args.get("filter_by_wfilter_upref", {}).get("current_user_profile", {})
+        results = []
```

Comment:
Minor: seems we don't need to define results outside the "with" statement.

Reply:
I initialized the `results` in case this code breaks and then we get `results` is not defined error. 
```
with session.begin_nested():
            session.execute("SET LOCAL enable_indexscan = OFF;")
            results = session.execute(text(sql), params) 
```

Reply:
If this code breaks, you won't hit this line ```for row in results``` unless you put it into a try-catch. But I think it is too defensive for programming in this way. So, it looks like is redundant. But again, it's minor. You don't have to change it.